src/SourceBuild/content/eng/join-verticals.proj

```diff
@@ -0,0 +1,44 @@
+<Project Sdk="Microsoft.Build.Traversal">
+
+    <PropertyGroup>
+      <TargetFramework>$(NetCurrent)</TargetFramework>
+    </PropertyGroup>
+  
+    <ItemGroup>
+      <ProjectReference Include="$(TasksDir)Microsoft.DotNet.UnifiedBuild.Tasks\Microsoft.DotNet.UnifiedBuild.Tasks.csproj" BuildInParallel="true" />
+    </ItemGroup>
+
+    <Target Name="BuildTasksProject">
+      <MSBuild Projects="$(TasksDir)Microsoft.DotNet.UnifiedBuild.Tasks\Microsoft.DotNet.UnifiedBuild.Tasks.csproj" Targets="Build" />
+    </Target>
+
+    <UsingTask TaskName="Microsoft.DotNet.UnifiedBuild.Tasks.JoinVerticals" AssemblyFile="$(MicrosoftDotNetUnifiedBuildTasksAssembly)" TaskFactory="TaskHostFactory" />
+    
+    <Target Name="JoinVerticals" DependsOnTargets="BuildTasksProject" AfterTargets="BuildTasksProject" >
```

Comment:
See the other comment in this file. Condition this target to depend on P2Ps being resolved (meaning built) and also sequence this target to actually run as part of the Build target.

```suggestion
    <Target Name="JoinVerticals" DependsOnTargets="ResolveProjectReferences" AfterTargets="Build">
```